airflow/providers/amazon/aws/hooks/base_aws.py

```diff
@@ -796,7 +796,10 @@ def test_connection(self):
         """
         try:
             session = self.get_session()
-            conn_info = session.client("sts").get_caller_identity()
+            test_endpoint_url = self.conn_config.extra_config.get('test_endpoint_url') or None
```

Comment:
By default, `get` returns None if the key does not exist.

```suggestion
            test_endpoint_url = self.conn_config.extra_config.get('test_endpoint_url')
```